src/components/link-as-button/index.tsx

```diff
@@ -0,0 +1,32 @@
+import * as React from 'react'
+import { ButtonSize } from '../button'
+import styled from 'styled-components'
+import { applyLinkStyles } from '../link/styles'
+import { StyledLinkProps } from '../link'
+
+/**
+ * The component is used where we need to have
+ * a Link which behaves AS a Button
+ */
+
+const StyledLink = styled.button<StyledLinkProps>`
+  ${(p) => applyLinkStyles({ size: p.size, disabled: p.disabled })}
+`
```

Comment:
```suggestion
const StyledLink = Link.withComponent('button');
```

Reply:
Tady je trochu problém, že `withComponent` má v dokumentaci použito, že se jedná o kandidáta na deprekování ([zdroj](https://styled-components.com/docs/api#withcomponent)). 

Přemýšlel jsem i nad použítím `as` a vzhledem k props mě přijde rozdělení do 4 komponent jako přijatelná varianta, která mne neuráží. 

Nicméně bych komponenty zabalil do nějakého kompaktního modulu. 

cc @fidelman

Reply:
To je dobre info ze planuju withComponent deprecovat. Potom by mozno dobre riesenie bolo pouzit nieco na tento sposob.
```ts
const StyledLink = Link as /* pretypovat aby sa componenta tvarila ako button elem*/;
StyledLink.defaultProps = {
  as: 'button'
};
```